kafka-connect-s3/docs/s3_connector.rst

```diff
@@ -95,6 +95,21 @@ NO compatibility, BACKWARD compatibility, FORWARD compatibility and FULL compati
 
 Schema evolution in the S3 connector works in the same way as in the `HDFS connector <../../../connect-hdfs/docs/hdfs_connector.html#schema-evolution>`_.
   
+Automatic Retries
+-----------------
+Users' environments might experience network partitioning, reach AWS throttling limits etc.
+In order to have a robust system it makes sense to have a retry policy, however, it's important
+to avoid a thundering herd problem.
+The S3 connector uses retry policy, predefined by ``com.amazonaws.retry.PredefinedRetryPolicies.SDKDefaultRetryCondition``
+and ``com.amazonaws.retry.PredefinedBackoffStrategies.FullJitterBackoffStrategy`` provided by AWS SDK.
+
+To calculate a maximum delay time before a next attempt, following formula ``${retry.backoff
+.ms} * 2 ^ (retry-1)`` is used.
```

Comment:
Maybe:
> The delay for retries is dependent upon the connector's ``retry.backoff.ms`` and ``s3.part.retries`` configuration properties. The actual delay is randomized, but the maximum delay can be calculated as a function of the number of retry attempts with ``${retry.backoff.ms} * 2 ^ (retry-1)``. In order to keep the maximum delay within a reasonable duration, it is capped at 24 hours.

How does `s3.part.retries` play into this?

Reply:
``s3.part.retries`` defines max number of attempts. 
The real number of attempts might be lesser since AWS client has some internal logic to determine whether it should retry request or not. One example - ``com.amazonaws.util.CapacityManager`` used to keep track of retries of failed requests(except throttle failures), so once capacity if drained - failed request won't be retried even if a number of attempts is lesser than ``s3.part.retries``.
Having all said above, I feel like we shouldn't dive that deep into internals but rather mention a high-level description of ``s3.part.retries``.